Framework/Algorithms/src/MaxEnt.cpp

```diff
@@ -229,17 +251,54 @@ std::map<std::string, std::string> MaxEnt::validateInputs() {
 
   MatrixWorkspace_sptr inWS = getProperty("InputWorkspace");
 
+  size_t nHistograms = 0;
   if (inWS) {
     // If the input signal is complex, we expect an even number of histograms
     // in the input workspace
 
-    size_t nhistograms = inWS->getNumberHistograms();
+    nHistograms = inWS->getNumberHistograms();
     bool complex = getProperty("ComplexData");
-    if (complex && (nhistograms % 2))
+    if (complex && (nHistograms % 2))
       result["InputWorkspace"] = "The number of histograms in the input "
                                  "workspace must be even for complex data";
+    if (!complex)
+      nHistograms *=
+          2; // Double number of histograms to compare with adjustments.
   }
 
+  // Check linear adjustments, we expect and even number of histograms
+  // and if any, they must be sufficient for all spectra in input workspace,
+  // if per spectrum reconstruction is done.
+  bool psr = getProperty("perSpectrumReconstruction");
```

Comment:
can we have a more descriptive name than psr?

Reply:
I haven't thought of one not almost as long as "perSpectrumReconstruction" and I thought "psr" would be OK for a short scope.

Reply:
Should perSpectrumReconstruction be PerSpectrumReconstruction?